crypto/asn1/d2i_pu.c

```diff
@@ -32,7 +32,7 @@ EVP_PKEY *d2i_PublicKey(int type, EVP_PKEY **a, const unsigned char **pp,
     } else
         ret = *a;
 
-    if (!EVP_PKEY_set_type(ret, type)) {
+    if (type != EVP_PKEY_id(ret) && !EVP_PKEY_set_type(ret, type)) {
```

Comment:
The documentation for `d2i_PublicKey` should be updated to explain this requirement.

Reply:
Done